src/routes/import.rs

```diff
@@ -0,0 +1,33 @@
+use axum::{extract::Query, response::IntoResponse};
+use axum_extra::extract::CookieJar;
+use reqwest::StatusCode;
+use serde::Deserialize;
+use tracing::instrument;
+use uuid::Uuid;
+
+use crate::middlewares::Schedule;
+
+use super::AppError;
+
+#[derive(Deserialize)]
+pub struct Params {
+    blob: String,
+}
+
+#[instrument(level = "debug")]
+pub async fn get(Query(Params { blob }): Query<Params>) -> Result<impl IntoResponse, AppError> {
+    let uuid = Uuid::new_v4().to_string();
+
+    Schedule::try_from(blob)
+        .map(|s| {
+            (
+                StatusCode::MOVED_PERMANENTLY,
```

Comment:
It might be worth using [302 found](https://www.rfc-editor.org/rfc/rfc9110#name-302-found) instead to allow importing the same schedule multiple times.

Reply:
Ah I used the `cache-control` header to make sure the browser doesn't keep it, but 302 is better. I'll update and get you to click approve again